FC-0068 docs: Documentation Templates (How-to, Reference, Concept) (#619)
openedx/docs.openedx.org: source/documentors/references/templates/doc_quickstart_template.rst

> Quickstrat Document Template

Reviewer: ```suggestion
Quickstart Document Template
```
Be sure you're using an English spellchecker on these documents

############################

What Is a Quickstart
********************
A quickstart is a brief, streamlined guide that helps users get up and running with a tool, project, or system as quickly as possible. Quickstarts focus on the essential steps to start working with a particular technology, often covering:

#. **Setup Requirements** – Prerequisites like software dependencies or configuration steps.

#. **Installation** – How to install the necessary files or dependencies.

#. **Primary Usage** – Commands or actions to perform basic operations.

#. **Testing and Verification** – Ways to ensure everything is working correctly.

#. **Common Next Steps** – Information on where to go next, often with links to detailed documentation.

> A good quickstart is designed for users who want to dive in with minimal setup. It benefits developers, allowing them to try out a tool or platform without wading through extensive documentation upfront. 

Reviewer: ```suggestion
A good quickstart is designed for users who want to dive in with minimal setup. For example, allowing developers to try out a tool or platform without wading through extensive documentation upfront, or empowering course authors to quickly get up and running with a complex area of Studio.
```

To call out that Quickstarts are not _just_ for developers, which is how this currently reads. Quickstarts are for all personas.

Prerequisites Example
**********************

- Outline required knowledge (e.g., familiarity with Open edX basics).
- List any tools or configurations needed (e.g., access to an Open edX instance, software, or plugins).

How To Embed One Document in Another
************************************

A Quickstart may be the union of different documents, such as how-tos, references, and concepts. To avoid duplicate documents, you can embed one document into another using the ``include`` directive. This directive allows you to include the content of another file located in the same repository. Here's how you can do it:

1. Copy the following code in the file to embed the other document.

.. code-block:: RST

	.. include:: path/to/other_file.rst

2. Replace the example path with the embedded document path. For example:

.. code-block:: RST

	include:: ../templates/doc_how_to_template.rst

How to Embed Only a Section
***************************

If you want to include or embed only certain sections of a document, you can follow these steps:

1.  Open the document you want to embed. Add ``.. START HERE`` at the beginning and ``.. END HERE`` at the end of the section you want to embed.

> 2. You will embed the other document in the document. Use the following code to the exact place where you want to add that particular section. Do not forget to edit the document link that you want to include in your Quickstart

Reviewer 2: Add a final stop after "Quickstart".

.. code-block:: RST

	.. include:: docs/section1.rst
  	:start-after: .. START HERE
   	:end-before: .. END HERE

The reader will see something like this:

.. include:: ../templates/doc_how_to_template.rst
 	:start-after: .. START HERE
 	:end-before: .. END HERE

Useful Resources
****************

> Here is a list of the most used resources to create a Quickstart

Reviewer 2: Add a final stop after "Quickstart".

> Here is a list of the most used resources to create a Quickstart

Reviewer: ```suggestion
Here is a list of the most used resources to create a Quickstart:
```

Images
======

Add the images and audiovisual resources you think could be necessary to clarify the process. Remember that the callouts should be in Open edX red ``(#d23228)`` and round-edged rectangles. 

.. code-block:: RST

	.. image:: /_images/documentors_howto/image_example.png
		:width: 500px
		:align: center
		:alt: This image example shows how to access the GitHub repository from the Open edX documentation repository.

.. image:: /_images/documentors_howto/image_example.png
		:width: 500px
		:align: center
		:alt: This image example shows how to access the GitHub repository from the Open edX documentation repository.

Links:
======

External Links
--------------

> You can have `inline links <https://docs.openedx.org/en/latest/documentors/references/quick_reference_rst.html#linking>`_ and `indirect links`_, that are useful if you want to link to the same thing multiple times, or if the URL is really long and you want things to read more cleanly.

Reviewer: ```suggestion
You can use `inline links <https://docs.openedx.org/en/latest/documentors/references/quick_reference_rst.html#linking>`_ or `indirect links`_. Indirect links are useful if you want to link to the same thing multiple times, or if the URL is really long and you want the raw RST to read more cleanly.
```

.. _Indirect links: https://docs.openedx.org/en/latest/documentors/references/quick_reference_rst.html#linking

Directives:
===========

Usually, for this type of document, you can use different directives like ``warning``, ``note``,  and ``see also``, among others.

> ..  see also:: Review the Style Guide to learn more about directives and other resources to create your documentation.

Reviewer 2: Should we add the link to the Style Guide, right?

Reviewer: Yes, please link to the Style Guide

Reviewer: And please add this See Also to all 4 template guides.

	**********************

	- Outline required knowledge (e.g., familiarity with Open edX basics).
	- List any tools or configurations needed (e.g., access to an Open edX instance, software, or plugins).

	How To Embed One Document in Another
	************************************

	A Quickstart may be the union of different documents, such as how-tos, references, and concepts. To avoid duplicate documents, you can embed one document into another using the ``include`` directive. This directive allows you to include the content of another file located in the same repository. Here's how you can do it:

	1. Copy the following code in the file to embed the other document.

	.. code-block:: RST

		.. include:: path/to/other_file.rst

	2. Replace the example path with the embedded document path. For example:

	.. code-block:: RST

		include:: ../templates/doc_how_to_template.rst

	How to Embed Only a Section
	***************************

	If you want to include or embed only certain sections of a document, you can follow these steps:

	1.  Open the document you want to embed. Add ``.. START HERE`` at the beginning and ``.. END HERE`` at the end of the section you want to embed.

> 	2. You will embed the other document in the document. Use the following code to the exact place where you want to add that particular section. Do not forget to edit the document link that you want to include in your Quickstart

Reviewer 2: Final stop after "Quickstart".

	.. code-block:: RST

		.. include:: docs/section1.rst
	  	:start-after: .. START HERE
	   	:end-before: .. END HERE

	The reader will see something like this:

	.. include:: ../templates/doc_how_to_template.rst
	 	:start-after: .. START HERE
	 	:end-before: .. END HERE

	Useful Resources
	****************

	Here is a list of the most used resources to create a Quickstart

	Images
	======

	Add the images and audiovisual resources you think could be necessary to clarify the process. Remember that the callouts should be in Open edX red ``(#d23228)`` and round-edged rectangles. 

	.. code-block:: RST

		.. image:: /_images/documentors_howto/image_example.png
			:width: 500px
			:align: center
			:alt: This image example shows how to access the GitHub repository from the Open edX documentation repository.

	.. image:: /_images/documentors_howto/image_example.png
			:width: 500px
			:align: center
			:alt: This image example shows how to access the GitHub repository from the Open edX documentation repository.

	Links:
	======

	External Links
	--------------

> 	You can have `inline links <https://docs.openedx.org/en/latest/documentors/references/quick_reference_rst.html#linking>`_ and `indirect links`_, that are useful if you want to link to the same thing multiple times, or if the URL is really long and you want things to read more cleanly.

Reviewer: ```suggestion
	You can use `inline links <https://docs.openedx.org/en/latest/documentors/references/quick_reference_rst.html#linking>`_ or `indirect links`_. Indirect links are useful if you want to link to the same thing multiple times, or if the URL is really long and you want the raw RST to read more cleanly.
```

Usually, for this type of document, you can use different directives like ``warning``, ``note``,  and ``see also``, among others.

> ..  see also:: Review the `Style Guide <https://docs.openedx.org/en/latest/documentors/references/doc_style_guide.html>`_ to learn more about directives and other resources to create your documentation.

Reviewer: This isn't rendering in the output

<img width="811" alt="image" src="https://github.com/user-attachments/assets/ffe6c8ee-b951-4a1d-9e22-0d7ab41687ea">

https://docsopenedxorg--619.org.readthedocs.build/en/619/documentors/references/templates/doc_quickstart_template.html
